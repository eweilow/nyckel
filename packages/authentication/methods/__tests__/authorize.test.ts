import { authorizeUser, GlobalAuthenticationConfig } from "../..";

jest.mock("../../state/generate", () => {
  return {
    generateCSRFPair: jest.fn(() => {
      return {
        secret: "secret",
        token: "token"
      };
    })
  };
});

const config: GlobalAuthenticationConfig = {
  audience: "param:audience",
  authorizationDomain: "http://param:authorizationDomain.com",
  clientId: "param:clientId",
  clientSecret: "param:clientSecret",
  urls: {
    authorization: "url:authorization",
    issuer: "url:issuer",
    jwks: "url:jwks",
    logout: "url:logout",
    token: "url:token",
    userinfo: "url:userinfo"
  }
};

describe("authorizeUser", () => {
  it("returns correctly", async () => {
    const data = await authorizeUser(
      ["scopeA", "scopeB"],
      "redirectUri",
      config
    );
    expect(data).toMatchSnapshot();
  });
});
