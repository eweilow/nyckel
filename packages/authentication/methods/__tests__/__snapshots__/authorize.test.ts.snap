// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authorizeUser returns correctly 1`] = `
Object {
  "csrfPair": Object {
    "secret": "secret",
    "token": "token",
  },
  "redirectTo": "url:authorization?response_type=code&client_id=param%3AclientId&redirect_uri=redirectUri&scope=scopeA%20scopeB%20openid%20profile%20email%20offline_access&state=token&audience=param%3Aaudience",
}
`;
